src/main/java/fi/otavanopisto/kuntaapi/server/controllers/IdentifierController.java

```diff
@@ -214,12 +223,28 @@ public void deleteIdentifier(Identifier identifier) {
     identifierDAO.delete(identifier);
   }
 
-  private Identifier createIdentifier(Long orderIndex, String type, String kuntaApiId, String source, String sourceId, String organizationKuntaApiId) {
-    return identifierDAO.create(orderIndex, type, kuntaApiId, source, sourceId, organizationKuntaApiId);
+  private Identifier createIdentifier(Identifier parent, Long orderIndex, String type, String kuntaApiId, String source, String sourceId, String organizationKuntaApiId) {
+    return identifierDAO.create(parent, orderIndex, type, kuntaApiId, source, sourceId, organizationKuntaApiId);
   }
-
-  public Identifier updateIdentifierOrderIndex(Identifier identifier, Long orderIndex) {
-    return identifierDAO.updateOrderIndex(identifier, orderIndex);
+  
+  public Identifier updateIdentifier(Identifier identifier, BaseId parentId, Long orderIndex) {
+    Identifier parent = null;
+    if (parentId != null) {
+      parent = findIdentifierById(parentId);
+      if (parent == null) {
+        logger.severe(String.format("Could not find parent %s for identifier %s", parentId, identifier.getKuntaApiId()));
+      }
+    }
+    
+    return updateIdentifier(identifier, parent, orderIndex);
+  }
+  
+  private Identifier updateIdentifier(Identifier identifier, Identifier parent, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateIdentifier" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)